apps/prairielearn/assets/scripts/instructorFileEditorClient.tsx

```diff
@@ -100,6 +176,156 @@ class InstructorFileEditor {
         .querySelector<HTMLButtonElement>('.js-reformat-file')
         ?.addEventListener('click', () => this.reformatJSONFile());
     }
+
+    // Override the save button click to show confirmation modal if needed
+    this.saveElement?.addEventListener('click', (e) => this.handleSaveClick(e));
+  }
+
+  /**
+   * Handles the save button click, showing a confirmation modal if there are issues,
+   * or proceeding with the save if the user has already confirmed.
+   */
+  handleSaveClick(event: MouseEvent) {
+    if (this.confirmedSave) {
+      // User has confirmed the save, so we need to restore the original UUID
+      this.restoreOriginalUuid();
+      this.confirmedSave = false;
+      return;
+    }
+
+    const errorResult = this.checkForSaveIssues();
+
+    if (errorResult) {
+      event.preventDefault();
+      this.showConfirmationModal(errorResult);
+    }
+
+    // Otherwise, continue with the save
+  }
+
+  /**
+   * Restores the original UUID in the editor contents when the user confirms save.
+   */
+  restoreOriginalUuid() {
+    if (!this.fileMetadata?.uuid) return;
+
+    const currentContents = this.editor.getValue();
+    const parsedContent = JSON.parse(currentContents);
+    if (typeof parsedContent === 'object' && parsedContent !== null) {
+      // Restore the original UUID
+      parsedContent.uuid = this.fileMetadata.uuid;
+      const restoredContents = JSON.stringify(parsedContent, null, 2);
+      this.editor.setValue(restoredContents);
+    }
+  }
+
+  /**
+   * Checks for issues that would require confirmation before saving.
+   *
+   * @returns An error result object or null if there are no issues.
+   */
+  checkForSaveIssues(): {
+    errorCode: SaveErrorCode;
+    originalUuid?: string;
+    newUuid?: string;
+  } | null {
+    const currentContents = this.editor.getValue();
+
+    if (this.fileMetadata && this.fileMetadata.type !== FileType.File) {
+      try {
+        const parsedContent = JSON.parse(currentContents);
+
+        if (typeof parsedContent !== 'object' || parsedContent == null) {
+          return { errorCode: SaveErrorCode.INVALID_JSON };
+        } else if (this.fileMetadata.uuid) {
```

Comment:
This line and the behavior of `getFileMetadataForPath(...)` means that we won't complain if a UUID is removed from an `infoCourse.json` file. Is this intentional? I guess this case is a little different from non-course JSON files because course UUIDs don't mean much (outside of "is this the example course") and aren't persisted anywhere.